sdks/python/apache_beam/runners/dataflow/dataflow_runner.py

```diff
@@ -430,6 +430,15 @@ def _check_for_unsupported_fnapi_features(self, pipeline_proto):
                 components.coders[windowing_strategy.window_coder_id].spec.urn,
                 windowing_strategy.window_fn.urn))
 
+  def _adjust_types_for_dataflow(self, pipeline):
+    # Dataflow runner requires a KV type for GBK inputs, hence we enforce that
+    # here.
+    pipeline.visit(self.group_by_key_input_visitor())
+
+    # Dataflow runner requires output type of the Flatten to be the same as the
```

Comment:
@CraigChambersG I was under the impression that this wasn't true and that Dataflow v2 handled this.